src/probnum/filtsmooth/gaussfiltsmooth/gaussfiltsmooth.py

```diff
@@ -75,38 +77,51 @@ def filter(self, dataset, times, **kwargs):
                 **kwargs
             )
             rvs.append(filtrv)
-        return KalmanPosterior(times, rvs, self)
+        return KalmanPosterior(times, rvs, self, with_smoothing=False)
 
-    def smooth(self, filter_posterior, **kwargs):
+    def smooth(self, filter_posterior):
         """
         Apply Gaussian smoothing to a set of filtered means and covariances.
 
         Parameters
         ----------
         filter_posterior : KalmanPosterior
             Posterior distribution obtained after filtering
-        kwargs : ???
 
         Returns
         -------
         KalmanPosterior
             Posterior distribution of the smoothed output
         """
-        smoothed_rv = filter_posterior[-1]
-        locations = filter_posterior.locations
-        out_rvs = [smoothed_rv]
+        rv_list = self.smooth_list(filter_posterior, filter_posterior.locations)
+        return KalmanPosterior(
+            filter_posterior.locations, rv_list, self, with_smoothing=True
+        )
+
+    def smooth_list(self, rv_list, locations, final_rv=None):
+        """
+        Apply smoothing to a list of RVs with desired final random variable.
+
+        Specification of a final RV is useful to compute samples from a KalmanPosterior object,
+        because in this case, the final RV is a Dirac (over a sample from the final Normal RV)
+        and not a Normal RV.
```

Comment:
Oh and sorry to state that again but it could be a good idea to also format the docstring with `Parameters` and `Returns`, since the content matches that structure already anyways.